packages/flutter_tools/lib/src/fuchsia/fuchsia_kernel_compiler.dart

```diff
@@ -58,35 +58,30 @@ class FuchsiaKernelCompiler {
       '--filesystem-root', fsRoot,
       '--packages', '$multiRootScheme:///$relativePackagesFile',
       '--output', fs.path.join(outDir, '$appName.dil'),
-      // TODO(zra): Add back when this is supported again.
-      // See: https://github.com/flutter/flutter/issues/44925
-      // '--no-link-platform',
-      '--split-output-by-packages',
-      '--manifest', manifestPath,
       '--component-name', appName,
-    ];
 
-    if (buildInfo.isDebug) {
-      flags += <String>[
-        '--embed-sources',
-      ];
-    } else if (buildInfo.isProfile) {
-      flags += <String>[
-        '--no-embed-sources',
-        '-Ddart.vm.profile=true',
-        '--gen-bytecode',
-        '--drop-ast',
-      ];
-    } else if (buildInfo.isRelease) {
-      flags += <String>[
-        '--no-embed-sources',
-        '-Ddart.vm.release=true',
+      // AOT/JIT:
+      if (buildInfo.usesAot) ...<String>['--aot', '--tfa'],
+      if (!buildInfo.usesAot) ...<String>[
+        // TODO(zra): Add back when this is supported again.
+        // See: https://github.com/flutter/flutter/issues/44925
+        // '--no-link-platform',
+        '--split-output-by-packages',
+        '--manifest', manifestPath
+      ],
+
+      // debug, profile, jit release, release:
+      if (buildInfo.isDebug) '--embed-sources',
+      if (!buildInfo.isDebug) '--no-embed-sources',
+      if (buildInfo.isProfile) '-Ddart.vm.profile=true',
+      if (buildInfo.isRelease || buildInfo.isJitRelease) '-Ddart.vm.release=true',
+
+      // Not AOT and not debug then generate bytecode and drop AST.
+      if (!buildInfo.usesAot && !buildInfo.isDebug) ...<String>[
         '--gen-bytecode',
```

Comment:
I think we should be using gen-bytecode for release now. cc @alexmarkov to confirm

Reply:
Currently, we're not using bytecode in AOT configurations. When enabling it next time, we will flip the flag by default in kernel compiler, so this code doesn't need to change.

For Fuchsia, we're using bytecode in JIT/release mode, so we could change this condition to `if (buildInfo.isJitRelease)` to reflect this.

Reply:
Done.